client/components/tinymce/plugins/contact-form/plugin.js

```diff
@@ -23,26 +26,38 @@ const contactForm = editor => {
 	} );
 
 	editor.on( 'remove', () => {
-		ReactDom.unmountComponentAtNode( node );
+		unmountComponentAtNode( node );
 		node.parentNode.removeChild( node );
 		node = null;
 	} );
 
-	editor.addCommand( 'WP_ContactForm', () => {
+	editor.addCommand( 'wpcomContactForm', content => {
+		store.dispatch( initContactForm( deserialize( content ) ) );
+
 		function onClose() {
 			editor.focus();
 			renderModal( 'hide' );
 		};
 
 		function renderModal( visibility = 'show' ) {
-			ReactDom.render(
-				React.createElement( ContactFormDialog, {
-					showDialog: visibility === 'show',
-					onClose,
-					onInsertMedia( markup ) {
-						editor.execCommand( 'mceInsertContent', false, markup );
-					}
-				} ),
+			render(
+				createElement( Provider, { store },
+					createElement( ContactFormDialog, {
+						showDialog: visibility === 'show',
+						onAdd() {
+							store.dispatch( addDefaultField() )
+						},
+						onRemove( index ) {
+							store.dispatch( removeField( index ) );
+						},
+						onClose,
```

Comment:
Any reason you chose to reference this by name rather than inline-ing in the other props? Plan to use `onClose` outside the context of `render` in the future?